Lib/test/_test_multiprocessing.py

```diff
@@ -4680,6 +4682,89 @@ def test__all__(self):
         # Just make sure names in blacklist are excluded
         support.check__all__(self, multiprocessing, extra=multiprocessing.__all__,
                              blacklist=['SUBDEBUG', 'SUBWARNING'])
+
+
+class _TestCustomReducer(BaseTestCase):
+
+    ALLOWED_TYPES = ('processes',)
+
+    def setUp(self):
+        self.original_reducer = multiprocessing.get_reducer()
+        spy = unittest.mock.MagicMock(spec_set=self.original_reducer.ForkingPickler,
+                                      wraps=self.original_reducer.ForkingPickler)
+        self.spy = spy
+
+        class SpyReducer(reduction.AbstractReducer):
+            def get_pickler_class(self):
+                return spy
+
+        multiprocessing.set_reducer(SpyReducer())
```

Comment:
I'd like to see another similar test, but using a custom context rather than the global `multiprocessing.set_reducer` hook. I suspect it might not work as expected, i.e. the processes and queues created with this custom context would not use the custom pickler.